eth/db/__init__.py

```diff
@@ -7,21 +7,22 @@
 
 from eth_utils import import_string
 
-from eth.db.backends.base import BaseAtomicDB
+from eth.abc import AtomicDatabaseAPI
 
 
 DEFAULT_DB_BACKEND = 'eth.db.atomic.AtomicDB'
 
 
-def get_db_backend_class(import_path: str = None) -> Type[BaseAtomicDB]:
+def get_db_backend_class(import_path: str = None) -> Type[AtomicDatabaseAPI]:
     if import_path is None:
         import_path = os.environ.get(
             'CHAIN_DB_BACKEND_CLASS',
             DEFAULT_DB_BACKEND,
         )
-    return cast(Type[BaseAtomicDB], import_string(import_path))
+    return cast(Type[AtomicDatabaseAPI], import_string(import_path))
 
 
-def get_db_backend(import_path: str = None, **init_kwargs: Any) -> BaseAtomicDB:
+def get_db_backend(import_path: str = None, **init_kwargs: Any) -> AtomicDatabaseAPI:
     backend_class = get_db_backend_class(import_path)
-    return backend_class(**init_kwargs)
+    # mypy doesn't understand the constructor  of AtomicDatabaseAPI
```

Comment:
Does it work if you add back the explicit `@abstractmethod` with `__init__` that was in `BaseDB`?

```py
    @abstractmethod	
    def __init__(self) -> None:	
        raise NotImplementedError(	
            "The `init` method must be implemented by subclasses of BaseDB"	
        )
```

Reply:
Yes, but I don't believe there is a generic `__init__` method for `AtomicDatabaseAPI` implementations.  We have one right now that wraps another database but I don't think that's the only constructor.

Also, I'm inclined to remove this functionality entirely because I don't think we actually use it in any meaningful way.